### PR TITLE
Port forwarding: UI

### PR DESCRIPTION
**Pull request by mook-as**
This is the first go of the UI for managing port forwards, for #53.

This is based on #106 and should not be merged before that one.

Screenshot:
image

The port forwarding table might be too big to fit in the Kubernetes Settings screen; we might want to break it out into a separate one.  The vertical scrollbar, especially when there's non-table things on the same screen, looks kind of bad.

We may also want to attempt to prevent people from manually unhooking the port forwarding for the dashboard.

### CODE REVIEW

File: src/plugins/i18n.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import Vue from 'vue';
```
Comment (ericpromislow) [inline src/plugins/i18n.js:1]: I'm wondering if pulling in i18n from dashboard should be a separate commit. There's a lot going on in both of these, and it would be easier to understand each on its own.

File: src/store/prefs.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+// Keys must be lowercase and valid dns label (a-z 0-9 -)
+export const CLUSTER = create('cluster', '');
+export const LAST_NAMESPACE = create('last-namespace', '');
+export const NAMESPACE_FILTERS = create('ns', ['all://user'], { parseJSON });
+export const WORKSPACE = create('workspace', '');
+export const EXPANDED_GROUPS = create('open-groups', ['cluster', 'rbac', 'serviceDiscovery', 'storage', 'workload'], { parseJSON });
+export const FAVORITE_TYPES = create('fav-type', [], { parseJSON });
+export const GROUP_RESOURCES = create('group-by', 'namespace');
+export const DIFF = create('diff', 'unified', { options: ['unified', 'split'] });
+export const THEME = create('theme', 'auto', {
```
Comment (ericpromislow) [inline src/store/prefs.js:51]: Well if we're pulling in this code I guess we can implement three-way dark mode (always dark, always light, do whatever the system mode is).

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Just to reiterate, not just prefs, but could you split this PR into two, where the first brings in all those dashboard modules, and the second does the actual port-forwarding?

**Comment by ericpromislow** [comment]
From #106 (where it didn't belong):

The i18n module isn't hooked up. When I'm not connected to a cluster I see this text in an empty sortableTable:

%sortableTable.noRows%
The translation table has an entry for it: There are no rows to show.

File: src/config/private-label.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+export const ANY = 0;
+export const STANDARD = 1;
+export const CUSTOM = 2;
+
+let mode = STANDARD;
+let vendor = 'Rancher';
+let product = 'Dashboard';
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/config/private-label.js:7]: That was 236b21b3439a79d029404f61d3147523f93ac80f (but I'll probably do that within the first step PR to pull in dashboard things).

**Review by mattfarina — DISMISSED** [review]
Can we move the port forwarding UI off of the k8s settings page. There is just so much on this preferences screen.

**Comment by mook-as** [comment]
I'll do that once #106 is merged; doesn't make much sense to rebase this again until then.

File: src/utils/object.js [hunk 4]
```diff
@@ -120,6 +120,8 @@ export function cleanUp(obj) {
             return cleaned;
           }
         }
+
+        return false;
```
Comment (ericpromislow) [inline src/utils/object.js:124]: As opposed to an implicit `return undefined`?

File: src/utils/object.js [hunk 5]
```diff
@@ -205,6 +207,8 @@ export function nonEmptyValueKeys(obj) {
     } else if (!!val || val === false || val === 0) {
       return key;
     }
+
+    return null;
```
Comment (ericpromislow) [inline src/utils/object.js:211]: again as opposed to an implicit `return undefined`?

Are you going to submit a PR to dashboard?

**Review by ericpromislow — APPROVED** [review]
Couple of questions and a suggestion to make an upstream PR, but otherwise the code looks good and it works.

File: src/utils/object.js (hunk 4 above)
Comment (mook-as) [reply src/utils/object.js]: Commit dropped, since we updated to dashboard's eslint configs and this no longer shows up :)

File: src/utils/object.js (hunk 5 above)
Comment (mook-as) [reply src/utils/object.js]: Commit dropped :)
